src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -109,6 +114,49 @@ TEST_F(AddSubMenuRequestTest, OnEvent_SUCCESS) {
   EXPECT_EQ(mobile_apis::Result::SUCCESS, kReceivedResult);
 }
 
+TEST_F(AddSubMenuRequestTest,
+       OnEvent_SubMenuWithIcon_NotExistsInFolder_ExpectInfo) {
+  // message will be send on hmi with icon that not exists in folder
+
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
+  (*command_msg_)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg_)[am::strings::msg_params][am::strings::sub_menu_icon]
+                 [am::strings::value] = "icon.png";
+
+  Event event(hmi_apis::FunctionID::UI_AddSubMenu);
+  MessageSharedPtr event_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*event_msg)[am::strings::params][am::hmi_response::code] =
+      mobile_apis::Result::WARNINGS;
+  (*event_msg)[am::strings::msg_params] = 0;
+
+  event.set_smart_object(*event_msg);
+
+  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>(command_msg_));
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app));
+
+  EXPECT_CALL(mock_app_manager_, get_settings())
+      .WillOnce(ReturnRef(mock_app_manager_settings_));
+
+  const std::string file_storage = "test_storage";
+  EXPECT_CALL(mock_app_manager_settings_, app_storage_folder())
+      .WillOnce(ReturnRef(file_storage));
+
+  MessageSharedPtr result_msg(
+      CatchMobileCommandResult(CallOnEvent(*command, event)));
+  const mobile_apis::Result::eType kReceivedResult =
+      static_cast<mobile_apis::Result::eType>(
+          (*result_msg)[am::strings::msg_params][am::strings::result_code]
+              .asInt());
+
+  EXPECT_EQ(mobile_apis::Result::WARNINGS, kReceivedResult);
+  const std::string received_info =
+      (*result_msg)[am::strings::msg_params][am::strings::info].asString();
+  EXPECT_EQ("Reference image(s) not found", received_info);
```

Comment:
@VVeremjova 1) Don't compare `const char*` to `string` even that it will be implicitly converted.
2) Please create const string for "Reference image(s) not found"